examples/constant-force-optical-trap/force-bias-optical-trap.py

```diff
@@ -76,242 +89,270 @@ def construct_nonuniform_bins(x_n, nbins):
-    df_i = results['df_i']
+        # Compute statistical inefficiency for extension timeseries
+        g = timeseries.statistical_inefficiency(x_kt[k, 0 : T_k[k]], x_kt[k, 0 : T_k[k]])
+        # store statistical inefficiency
+        g_k[k] = g
+        print(
+            f"timeseries {k + 1:d} : g = {g:.1f}, {int(np.floor(T_k[k] / g)):.0f} "
+            f"uncorrelated samples (of {T_k[k]:d} total samples)"
+        )
+        N_max = max(N_max, int(np.ceil(T_k[k] / g)) + 1)
+
+    # Subsample trajectory position data.
+    x_kn = np.zeros([K, N_max])
+    bin_kn = np.zeros([K, N_max])
+    N_k = np.zeros([K], int)
+    for k in range(K):
+        # Compute correlation times for potential energy and chi timeseries.
+        indices = timeseries.subsample_correlated_data(x_kt[k, 0 : T_k[k]])
+        # Store subsampled positions.
+        N_k[k] = len(indices)
+        x_kn[k, 0 : N_k[k]] = x_kt[k, indices]
+        bin_kn[k, 0 : N_k[k]] = bin_kt[k, indices]
+
+    # Set arbitrarynp.zeros for external biasing potential.
+    x0_k = np.zeros([K])  # x position corresponding to zero of potential
+    for k in range(K):
+        x0_k[k] = x_kn[k, 0 : N_k[k]].mean()
+    print("x0_k = ", x0_k)
+
+    # Compute bias energies in units of kT.
+    # u_kln[k,l,n] is the reduced (dimensionless) relative potential energy of snapshot n from umbrella simulation k evaluated at umbrella l
+    u_kln = np.zeros([K, K, N_max])
+    for k in range(K):
+        for l in range(K):
+            # compute relative energy difference from sampled state to each other state
+            # U_k(x) = F_k x
+            # where F_k is external biasing force
+            # (F_k pN) (x nm) (pN /
+            # u_kln[k,l,0:N_k[k]] = - pN_nm_to_kT * (biasing_force_k[l] - biasing_force_k[k]) * x_kn[k,0:N_k[k]]
+            u_kln[k, l, 0 : N_k[k]] = -pN_nm_to_kT * biasing_force_k[l] * (
+                x_kn[k, 0 : N_k[k]] - x0_k[l]
+            ) + pN_nm_to_kT * biasing_force_k[k] * (x_kn[k, 0 : N_k[k]] - x0_k[k])
+
+    # DEBUG
+    start_time = time.time()
+
+    # Initialize MBAR.
+    print("Running MBAR...")
+    # TODO: method is not part of the MBAR signature
+    #       `method="adaptive"`
+    mbar = pymbar.MBAR(u_kln, N_k, verbose=True, relative_tolerance=1.0e-10)
+
+    # Compute unbiased energies (all biasing forces are zero).
+    # u_kn[k,n] is the reduced potential energy without umbrella restraints of snapshot n of umbrella simulation k
+    u_kn = np.zeros([K, N_max])
+    for k in range(K):
+        #    u_kn[k,0:N_k[k]] = - pN_nm_to_kT * (0.0 - biasing_force_k[k]) * x_kn[k,0:N_k[k]]
+        u_kn[k, 0 : N_k[k]] = 0.0 + pN_nm_to_kT * biasing_force_k[k] * (
+            x_kn[k, 0 : N_k[k]] - x0_k[k]
+        )
+
+    # Compute PMF in unbiased potential (in units of kT).
+    print("Computing PMF...")
+    # TODO: PMF computations is not part of MBAR now?
+    # pmf = pymbar.PMF(u_kn, )
+    results = mbar.compute_pmf(u_kn, bin_kn, NBINS)
```

Comment:
What's the equivalent of `mbar.compute_pmf(...)` in v4, @mrshirts?

Reply:
OK, this is fixed in most recent commit.